tdrs-backend/tdpservice/parsers/validators/base.py

```diff
@@ -0,0 +1,171 @@
+"""Base functions to be overloaded and composed from within the other validator classes."""
+
+import functools
+from .util import _is_empty
+
+
+def _handle_cast(val, cast):
+    return cast(val)
+
+
+def _handle_kwargs(val, **kwargs):
+    if 'cast' in kwargs and kwargs['cast'] is not None:
+        val = _handle_cast(val, kwargs['cast'])
+
+    return val
+
+
+def base_validator(makeValidator):
+    """Wrap validator funcs to handle kwargs."""
+    @functools.wraps(makeValidator)
+    def _validator(*args, **kwargs):
+        validator = makeValidator(*args, **kwargs)
+
+        def _validate(val):
+            val = _handle_kwargs(val, **kwargs)
+            return validator(val)
+
+        return _validate
+
+    return _validator
+
+
+@base_validator
+def isEqual(option, **kwargs):
+    """Return a function that tests if an input param is equal to option."""
+    return lambda val: val == option
+
+
+@base_validator
+def isNotEqual(option, **kwargs):
+    """Return a function that tests if an input param is not equal to option."""
+    return lambda val: val != option
```

Comment:
You should be able to only return the condition and have the decorator to evaluate the lambda function, that way you will not to return "lambda val:". 
Although, I also see @elipe17's point about definition of decorator and I agree that we are not following the decorator formal definition, but we are using it's ability to change a group of functions.

Reply:
https://github.com/raft-tech/TANF-app/pull/3122#discussion_r1717423952